### ISSUE TITLE
Different metrics

### ISSUE THREAD

**Issue by LucaDeGrandis** [UniModal4Reasoning/ChartVLM#15]
Hi! Thank you for the great work!
I am encountering a small issue while trying ChartVLM (both base and large). I have randomly selected 50 examples from each chart type, run inference, and computed the SCRM metric. The result was very far from what is written in the paper (some chart are better, some are worse). So I decided to test on the ring charts. I've used the entire sample with Chart-VLM base. Unfortunately, I still get different results. The average precision is visible in the attached image.

![image](https://github.com/user-attachments/assets/75dff166-45a0-433b-93be-19131c9e422f)

**Comment by BOBrown** [comment]
@LucaDeGrandis Hi, please perform the evaluation process using our [metric](https://github.com/UniModal4Reasoning/ChartVLM/blob/main/eval/eval_SE_ChartX.py). We employ different thresholds for different difficulties of types of charts,  shown [here](https://github.com/UniModal4Reasoning/ChartVLM/blob/f8805e0fce1139a43419e8488feda89fe0898d23/eval/eval_SE_ChartX.py#L27C5-L27C15). Besides, we use entity-replace for prediction shown [here](https://github.com/UniModal4Reasoning/ChartVLM/blob/f8805e0fce1139a43419e8488feda89fe0898d23/eval/eval_SE_ChartX.py#L52).

**Comment by LucaDeGrandis** [comment]
@BOBrown Hi, thank you for the quick reply.

I tried using your script ([here](https://github.com/UniModal4Reasoning/ChartVLM/blob/main/eval/eval_SE_ChartX.py)). The results are better but are still different.

The following is the raw output from the script you suggested:

```
2024-09-05 15:10:26,596 - INFO - infer_result:/content/infer_result_on_ChartX.json
2024-09-05 15:10:28,483 - INFO - *************** Performance *****************
2024-09-05 15:10:28,483 - INFO - rings
2024-09-05 15:10:28,483 - INFO - 

            -----------------------------------------------------------

            |  Metrics   |  Sim_threshold  |  Tolerance  |    Value    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.1664    |     

            |            |                 ----------------------------

            | mPrecison  |  0.5:0.05:0.95  |   slight    |    0.1664    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.1857    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.2000    |

            |            |                  ---------------------------

            | Precison   |       0.5       |   slight    |    0.2000    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.2214    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.1536    |

            |            |                  ---------------------------

            | Precison   |      0.75       |   slight    |    0.1536    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.1714    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.1500    |

            |            |                  ---------------------------

            | Precison   |       0.9       |   slight    |    0.1500    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.1679    |

            -----------------------------------------------------------

            |Precison(EM)|        1        |   strict    |    0.1500    |

            -----------------------------------------------------------

            
2024-09-05 15:10:28,484 - INFO - *************** Performance *****************
2024-09-05 15:10:28,484 - INFO - average
2024-09-05 15:10:28,484 - INFO - 

            -----------------------------------------------------------

            |  Metrics   |  Sim_threshold  |  Tolerance  |    Value    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.1664    |     

            |            |                 ----------------------------

            | mPrecison  |  0.5:0.05:0.95  |   slight    |    0.1664    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.1857    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.2000    |

            |            |                  ---------------------------

            | Precison   |       0.5       |   slight    |    0.2000    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.2214    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.1536    |

            |            |                  ---------------------------

            | Precison   |      0.75       |   slight    |    0.1536    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.1714    |

            -----------------------------------------------------------

            |            |                 |   strict    |    0.1500    |

            |            |                  ---------------------------

            | Precison   |       0.9       |   slight    |    0.1500    |

            |            |                  ---------------------------

            |            |                 |    high     |    0.1679    |

            -----------------------------------------------------------

            |Precison(EM)|        1        |   strict    |    0.1500    |

            -----------------------------------------------------------

            

```

**Comment by BOBrown** [comment]
@LucaDeGrandis Could you show the link to the model used for evaluation, and we will check it @renqiux0302 @sky-fly97

**Comment by LucaDeGrandis** [comment]
@BOBrown of course, the model I used is ChartVLM-base available on huggingface at [this link](https://huggingface.co/U4R/ChartVLM-base).

**Comment by renqiux0302** [comment]
@LucaDeGrandis Thanks for your questions.

We divide the ChartX benchmark into 4,848 validation samples and 1,152 test samples. Both the validation and test samples underwent rigorous manual quality inspection. The 4,848 validation samples are employed to evaluate the model performance during the fine-tuning process. We report the performance of all models on the 1,152 test samples.

The 4,848 validation samples are listed [here](https://drive.google.com/file/d/13jwSO8kaAnbPujECQK9x2QA_TXByzzYH/view).

The 1,152 test samples are listed [here](https://drive.google.com/file/d/1kOEi5Kca7WnFhBGyJlBIEtlgaIk004o0/view).

Apologies for the delayed response. We have conducted the result reproduction, and the evaluation log is provided [here](https://github.com/UniModal4Reasoning/ChartVLM/blob/main/eval/eval_result_SE_on_ChartX.log). You can reproduce our performance based on the [test samples](https://drive.google.com/file/d/1kOEi5Kca7WnFhBGyJlBIEtlgaIk004o0/view).

**Comment by LucaDeGrandis** [comment]
@renqiux0302 Thank you for your answer.

I don't recall the paper mentioning the split. Is there an error in the paper?
Does this mean that you used early stopping during training of ChartVLM?

**Comment by renqiux0302** [comment]
@LucaDeGrandis Sorry for any confusion caused by the initial version of the paper. We have since updated the arXiv version [(v2)](https://arxiv.org/abs/2402.12185) to include a more detailed description of the validation and test set split to ensure clarity.

Additionally, we did not employ an early stopping strategy during the training of ChartVLM. The training process for the base decoder was conducted for a total of 50 epochs, and for the auxiliary decoder, we trained for 5 epochs. 

Thanks again for your question, and we hope this clarification is helpful.

**Comment by LucaDeGrandis** [comment]
@renqiux0302 Thank you for the clarifications, they were very helpful. I will read the updated version of the paper.

Unfortunately, I don't have the time to repeat the experiment with the new test set in the short term but I believe that is the reason behind the differences in our results. I will close the issue for now.

Thanks again for the clarifications!
